File: selenium-java-framework/src/main/java/pages/InventoryPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class InventoryPage {
    WebDriver driver;

    By title = By.className("title");
    By addBackpack = By.id("add-to-cart-sauce-labs-backpack");
    By addBikeLight = By.id("add-to-cart-sauce-labs-bike-light");
    By cartLink = By.className("shopping_cart_link");

    public InventoryPage(WebDriver driver) {
        this.driver = driver;
    }

    public String getPageTitle() {
        return driver.findElement(title).getText();
    }

    public void addBackpackToCart() {
        driver.findElement(addBackpack).click();
    }

    public void addBikeLightToCart() {
        driver.findElement(addBikeLight).click();
    }

    public void openCart() {
        driver.findElement(cartLink).click();
    }
}

File: playwright-python-framework/pages/inventory_page.py
from utils.base_page import BasePage


class InventoryPage(BasePage):
    PAGE_TITLE = ".title"
    ADD_BACKPACK = "#add-to-cart-sauce-labs-backpack"
    ADD_BIKE_LIGHT = "#add-to-cart-sauce-labs-bike-light"
    SHOPPING_CART_LINK = ".shopping_cart_link"

    def get_page_title(self):
        return self.get_text(self.PAGE_TITLE)

    def add_backpack_to_cart(self):
        self.click(self.ADD_BACKPACK)

    def add_bike_light_to_cart(self):
        self.click(self.ADD_BIKE_LIGHT)

    def open_cart(self):
        self.click(self.SHOPPING_CART_LINK)